### PR TITLE
tpm_util: Replace a logger.error with an Exception in case of invalid…

### PR DESCRIPTION
**Pull request by stefanberger**
… signature

This fixes a possibly severe issue in 7.2.5 & 7.3.0.

### CODE REVIEW

**Comment by stefanberger** [comment]
@mheese This is the more severe issue.

**Review by maugustosilva — APPROVED** [review]

File: keylime/tpm/tpm_util.py [hunk 1]
```diff
@@ -157,8 +157,8 @@ def checkquote(
 
     try:
         verify(pubkey, signature, quote_digest, hashfunc)
-    except InvalidSignature:
-        logger.error("Invalid quote signature!")
+    except InvalidSignature as e:
+        raise Exception("Invalid quote signature!") from e
```
Comment (aplanas) [inline keylime/tpm/tpm_util.py:161]: At this point I wonder why to catch an specific exception to throw a generic one. Maybe is better to drop the try all together and bubble up the original exception

File: keylime/tpm/tpm_util_test.py [hunk 2]
```diff
@@ -60,6 +60,33 @@ def test_checkquote(self) -> None:
         except Exception as e:
             self.fail(f"checkquote failed with {e}")
 
+        try:
+            # flip bit
+            quoteblob[5] ^= 0x1
+            checkquote(aikblob, nonce, sigblob, quoteblob, pcrblob, "sha256")
+            self.fail(f"checkquote did not fail signature verification as expected")
```
Comment (aplanas) [inline keylime/tpm/tpm_util_test.py:67]: This assert is more clear: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises

File: keylime/tpm/tpm_util.py (hunk 1 above)
Comment (stefanberger) [reply keylime/tpm/tpm_util.py]: Fixed.

File: keylime/tpm/tpm_util_test.py (hunk 2 above)
Comment (stefanberger) [reply keylime/tpm/tpm_util_test.py]: Thanks. Fixed.

**Review by aplanas — APPROVED** [review]
The lint complain that now the import for InvalidSignature should be removed, but beside that LGTM

**Comment by stefanberger** [comment]
Do we want to open branches for 7.3.x and 7.2.5.x to apply this patch to ?

**Comment by stefanberger** [comment]
This patch fixes CVE-2023-3674.

**Comment by aplanas** [comment]
> CVE-2023-3674

Is this ID correct? Is the CVE public or is under embargo? In the last case I think should not be published after the release?

**Comment by stefanberger** [comment]
> > CVE-2023-3674
> 
> Is this ID correct? Is the CVE public or is under embargo? In the last case I think should not be published after the release?

This is the ID I received in the email. I know, compared to the 'other' one it is a much lower number.

**Comment by ansasaki** [comment]
> > CVE-2023-3674
> 
> Is this ID correct? Is the CVE public or is under embargo? In the last case I think should not be published after the release?

Yes, let's please stop disclosing security issues.

**Comment by stefanberger** [comment]
Delete my messages.

**Pull request merged**